File: cf/O.cpp
//
// Created by maxim on 25.11.2020.
//

#include <bits/stdc++.h>


using ll = long long;
using ld = long double;
using namespace std;

int main() {
    int k, n;
    cin >> k >> n;
    vector<vector<ll>> cls(k, vector<ll>());
    for (int i = 0, x, y; i < n; i++) {
        cin >> x >> y;
        x--;
        cls[x].push_back(y);
    }

    //M(M(y^2|x)) = M(y^2)

    ld ans = 0;

    for (int i = 0; i < k; i++) {
        ld sum_sq = 0;
        for (auto &y : cls[i]) {
            sum_sq += (1.0 * y / n) * y;
        }
        ans += sum_sq;
    }

    //M( M(y|x)^2)
    ld res = 0;
    for (int i = 0; i < k; i++) {
        if (cls.empty()) continue;
        ld sum = 0;
        for (auto &y : cls[i]) {
            sum += y;
        }
        res += sum / (1.0 * n * cls[i].size()) * sum;
    }

    cout << ans - res;


    return 0;
}

File: cf/O.py
if __name__ == '__main__':
    k = int(input())
    n = int(input())
    cls = [[] for _ in range(k)]
    for i in range(n):
        x, y = map(int, input().split())
        cls[x - 1].append(y)
    ans = 0
    for i in range(k):
        for y in cls[i]:
            ans += y * y
    ans /= n

    res = 0
    for i in range(k):
        if len(cls[i]) == 0:
            continue
        s = 0
        for y in cls[i]:
            s += y
        s /= n
        res += s * s * n / len(cls[i])
    print(ans - res)
